### PR TITLE
Replace private API encodeURI with public API SigV4_EncodeURI

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
Description
-----------
This PR removes the redundant private API `encodeURI` with public API `SigV4_EncodeURI` introduced in #89 .

Test Steps
-----------
```
cmake -S test -B build/ -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Debug -DUNITTEST=ON -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -DNDEBUG -DLOGGING_LEVEL_DEBUG=1'
cd build/
make coverage
cd ..
declare -a EXCLUDE=("\*test\*" "\*CMakeCCompilerId\*" "\*mocks\*")
echo ${EXCLUDE[@]} | xargs lcov --rc lcov_branch_coverage=1 -r build/coverage.info -o build/coverage.info
lcov --rc lcov_branch_coverage=1 --list build/coverage.info
```
Before applying the patch
```
Summary coverage rate:
  lines......: 99.8% (931 of 933 lines)
  functions..: 98.2% (54 of 55 functions)
  branches...: 100.0% (497 of 497 branches)
Reading tracefile build/coverage.info
                         |Lines       |Functions  |Branches    
Filename                 |Rate     Num|Rate    Num|Rate     Num
===============================================================
[/home/ubuntu/Temp/kar-rahul-aws/karahulx_SigV4/SigV4-for-AWS-IoT-embedded-sdk/source/]
sigv4.c                  |99.8%    886|98.0%    51| 100%    475
sigv4_quicksort.c        | 100%     47| 100%     4| 100%     22
===============================================================
                   Total:|99.8%    933|98.2%    55| 100%    497
```
After applying the patch
```
Summary coverage rate:
  lines......: 100.0% (933 of 933 lines)
  functions..: 100.0% (55 of 55 functions)
  branches...: 100.0% (497 of 497 branches)
Reading tracefile build/coverage.info
                         |Lines       |Functions  |Branches    
Filename                 |Rate     Num|Rate    Num|Rate     Num
===============================================================
[/home/ubuntu/Temp/kar-rahul-aws/karahulx_SigV4/SigV4-for-AWS-IoT-embedded-sdk/source/]
sigv4.c                  | 100%    886| 100%    51| 100%    475
sigv4_quicksort.c        | 100%     47| 100%     4| 100%     22
===============================================================
                   Total:| 100%    933| 100%    55| 100%    497
```

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
#89

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by tony-josi-aws — DISMISSED** [review]

File: test/unit-test/sigv4_utest.c [hunk 1]
```diff
@@ -681,15 +693,15 @@ void test_SigV4_GenerateHTTPAuthorization_Happy_Paths()
     returnStatus = SigV4_GenerateHTTPAuthorization( &params, authBuf, &authBufLen, &signature, &signatureLen );
     TEST_ASSERT_EQUAL( SigV4Success, returnStatus );
     TEST_ASSERT_EQUAL( SIGV4_HASH_MAX_DIGEST_LENGTH * 2U, signatureLen );
-    printf( "%.*s\n", authBufLen, authBuf );
+    printf( "%ld\n.*%s\n", authBufLen, authBuf );
```
Comment (aggarg) [inline test/unit-test/sigv4_utest.c:696]: What is the reason for this change?

Comment (kar-rahul-aws) [reply test/unit-test/sigv4_utest.c]: Updated as per discussion.

**Review by aggarg — APPROVED** [review]

**Comment by archigup** [comment]
One question, I don't see the encodeURI function definition being removed? Should it be removed and inlined into the public function?

**Review by archigup — APPROVED** [review]
